### PR TITLE
Refactor: Add Menu/MenuItem types

### LINKED ISSUE #15177: Add TypeScript types to Menu, MenuItem
This issue is for adding TypeScript types to the Menu and MenuItem components.

Please review the information in the parent issue, and read and follow the steps outlined in the [adding component types documentation](https://github.com/carbon-design-system/carbon/blob/main/docs/guides/adding-component-types.md#steps-to-provide-baseline-type-definitions-for-components).

### PR DESCRIPTION
**Pull request by mattborghi**
Closes #15177

Add Typescript types for Menu and MenuItem.

#### Changelog

**New**

- Add Typescript types for Menu and MenuItem.

**Changed**

- File extensions changed from `.js` to `.tsx`/`.ts`.

**Removed**

- N/A

#### Testing / Reviewing

- Storybook runs as before.
- Unit tests passing.

### CODE REVIEW

File: packages/react/src/components/Menu/Menu.tsx [hunk 1]
```diff
@@ -221,22 +286,41 @@ const Menu = React.forwardRef(function Menu(
       max - spacing - size,
     ];
 
-    const bestOption = options.find((option) => option !== false);
+    // cast to number since we have the fallback value.
```
Comment (andreancardona) [inline packages/react/src/components/Menu/Menu.tsx]: Could we clarify this comment?

**Review by andreancardona — APPROVED** [review]
Other than one small comment - looks good!

File: packages/react/src/components/Menu/Menu.tsx (hunk 1 above)
Comment (mattborghi) [reply packages/react/src/components/Menu/Menu.tsx]: Done!

**Review by tw15egan — APPROVED** [review]
LGTM 👍 ✅

**Pull request merged**